examples/core/rgcn/hetero_rgcn.py

```diff
@@ -0,0 +1,654 @@
+        {
+            "y_true": y_true[split_idx["train"]["paper"]],
+            "y_pred": y_pred[split_idx["train"]["paper"]],
+        }
+    )["acc"]
+    valid_acc = evaluator.eval(
+        {
+            "y_true": y_true[split_idx["valid"]["paper"]],
+            "y_pred": y_pred[split_idx["valid"]["paper"]],
+        }
+    )["acc"]
+    test_acc = evaluator.eval(
+        {
+            "y_true": y_true[split_idx["test"]["paper"]],
+            "y_pred": y_pred[split_idx["test"]["paper"]],
+        }
+    )["acc"]
+
+    return train_acc, valid_acc, test_acc
+
+
+def is_support_affinity():
+    # dgl supports enable_cpu_affinity since 0.9.1.
+    return dgl.__version__ >= "0.9.1"
+
+
+def main(args):
+    device = f"cuda:0" if th.cuda.is_available() else "cpu"
+
+    # Prepare the data.
+    g, labels, num_classes, split_idx, logger, train_loader = prepare_data(
+        args, device
+    )
+
+    # Create the embedding layer and move it to the appropriate device.
+    embed_layer = rel_graph_embed(g, 128).to(device)
+
+    # Initialize the entity classification model.
+    model = EntityClassify(g, 128, num_classes).to(device)
+
+    print(
+        "Number of embedding parameters: "
+        f"{sum(p.numel() for p in embed_layer.parameters())}"
+    )
+    print(
+        "Number of model parameters: "
+        f"{sum(p.numel() for p in model.parameters())}"
+    )
+
+    for run in range(args.runs):
+        try:
+            embed_layer.reset_parameters()
+            model.reset_parameters()
+        except:
+            # Old pytorch version doesn't support reset_parameters() API.
+            ##################################################################
+            # [Why we need to reset the parameters?]
+            # If parameters are not reset, the model will start with the
+            # parameters learned from the last run, potentially resulting
+            # in biased outcomes or sub-optimal performance if the model was
+            # previously stuck in a poor local minimum.
+            ##################################################################
+            pass
+
+        # `itertools.chain()` is a function in Python's itertools module.
+        # It is used to flatten a list of iterables, making them act as
+        # one big iterable.
+        # In this context, the following code is used to create a single
+        # iterable over the parameters of both the model and the embed_layer,
+        # which is passed to the optimizer. The optimizer then updates all
+        # these parameters during the training process.
+        all_params = itertools.chain(
+            model.parameters(), embed_layer.parameters()
+        )
+        optimizer = th.optim.Adam(all_params, lr=0.01)
+
+        # If CPU affinity is supported and the device is CPU, execute training
+        # with CPU affinity enabled.
+        if args.num_workers != 0 and device == "cpu" and is_support_affinity():
```

Comment:
I feel utilizing CPU affinity to improve performance is not in the scope of this example. Can we remove this part? @frozenbugs

Reply:
Yes, I prefer simple and easy to understand examples.

Reply:
Get